src/ripple/json/impl/json_value.cpp

```diff
@@ -443,77 +443,61 @@ Value::type () const
     return type_;
 }
 
-
-int
-Value::compare ( const Value& other )
+static
+int integerCmp (Int i, UInt ui)
 {
-    /*
-    int typeDelta = other.type_ - type_;
-    switch ( type_ )
-    {
-    case nullValue:
+    // All negative numbers are less than all unsigned numbers.
+    if (i < 0)
+        return -1;
 
-       return other.type_ == type_;
-    case intValue:
-       if ( other.type_.isNumeric()
-    case uintValue:
-    case realValue:
-    case booleanValue:
-       break;
-    case stringValue,
-       break;
-    case arrayValue:
-       delete value_.array_;
-       break;
-    case objectValue:
-       delete value_.map_;
-    default:
-       JSON_ASSERT_UNREACHABLE;
-    }
-    */
-    return 0;  // unreachable
+    // All unsigned numbers with bit 0 set are too big for signed integers.
+    if (ui & 0x8000)
+        return 1;
+
+    // Now we can safely compare.
+    return (i < ui) ? -1 : (i == ui) ? 0 : 1;
 }
 
-bool
-Value::operator < ( const Value& other ) const
+bool operator < (const Value& x, const Value& y)
 {
-    int typeDelta = type_ - other.type_;
-
-    if ( typeDelta )
-        return typeDelta < 0 ? true : false;
+    if (auto delta = x.type_ - y.type_)
+    {
+        if (x.type_ == intValue && y.type_ == uintValue)
+            delta = integerCmp (x.value_.int_, y.value_.uint_);
```

Comment:
Well, this isn't _that_ unreasonable.  :-)

What they are doing is computing a signum function for the value, and then using it to compute < or > - and I added some logic to that. In fact, let me call it `signum`...